### PR TITLE
Fixcode

### PR DESCRIPTION
**Pull request by rmorali**
Superb work!!!

### CODE REVIEW

File: app/models/fleet.rb [hunk 1]
```diff
@@ -12,8 +12,12 @@ class Fleet < ApplicationRecord
   delegate :name, to: :unit
   delegate :capacity, to: :unit
 
+  def facility?
```
Comment (lunks) [inline app/models/fleet.rb]: `delegate :name, :capacity, :facility, :weight, to: :unit`

File: app/models/fleet.rb [hunk 2]
```diff
@@ -22,38 +26,37 @@ def cargo
 
   def available_capacity
```
Comment (lunks) [inline app/models/fleet.rb]: You can use `inject` here it seems.

File: app/models/fleet.rb [hunk 3]
```diff
@@ -22,38 +26,37 @@ def cargo
...
-    if self.unit.weight * quantity > carrier.available_capacity
-      quantity = carrier.available_capacity  / self.unit.weight
+  def load_in(carrier, quantity)
+    if unit.weight * quantity > carrier.available_capacity
+      quantity = carrier.available_capacity / unit.weight
       return nil if quantity < 1
     end
     if self.quantity == quantity
-      self.update_attributes(carrier: carrier, destination: carrier.destination, arrives_in: carrier.arrives_in)
+      update_attributes(carrier: carrier, destination: carrier.destination, arrives_in: carrier.arrives_in)
```
Comment (lunks) [inline app/models/fleet.rb]: I prefer using `update` instead of `update_attributes`, it's shorter.

File: app/models/fleet.rb [hunk 4]
```diff
@@ -22,38 +26,37 @@ def cargo
...
+      left_behind = dup
       left_behind.quantity = self.quantity - quantity
       left_behind.save
-      self.update_attributes(quantity: quantity, carrier: carrier, destination: carrier.destination, arrives_in: carrier.arrives_in)
+      update_attributes(quantity: quantity, carrier: carrier, destination: carrier.destination, arrives_in: carrier.arrives_in)
     end
   end
 
-  def unload_from(carrier,quantity)
+  def unload_from(_carrier, quantity)
```
Comment (lunks) [inline app/models/fleet.rb]: It's weird you don't need the carrier here. Check if you're not forgetting anything.

File: app/models/squad.rb [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
-    else
-      self.update_attributes(metals: self.metals - value) unless value > self.metals
-    end
+  def debit_credits(value)
+    update_attributes(credits: credits - value) unless value > credits
   end
 
+  def debit_metals(value)
+    update_attributes(metals: metals - value) unless value > metals
+  end
```
Comment (lunks) [inline app/models/squad.rb]: 👏

Comment (lunks) [reply app/models/squad.rb]: Prefer `if` over `unless`, it's easier to read.
`if metals <= value`

File: app/services/build_facility.rb [hunk 6]
```diff
@@ -5,20 +5,19 @@ def initialize(facility, squad, planet)
...
   def ready_in
     Round.get_current.number + @facility.producing_time
   end
 
   def build!
-    Fleet.create(quantity: 1, unit: @facility, squad: @squad, planet: @planet, round: Round.get_current, ready_in: self.ready_in) if self.facility? && self.credits?
+    Fleet.create(quantity: 1, unit: @facility, squad: @squad, planet: @planet, round: Round.get_current, ready_in: ready_in) if valid?
   end
-
 end
```
Comment (lunks) [inline app/services/build_facility.rb]: 👏

File: app/services/order_movement.rb [hunk 7]
```diff
@@ -1,25 +1,25 @@
...
-  def initialize(fleet,quantity,destination)
+  def initialize(fleet, quantity, destination)
     @fleet = fleet
     @quantity = quantity
     @destination = destination
-    self.cancel_move! if @quantity == 0 || @destination == nil
+    cancel_move! if @quantity.zero? || @destination.nil?
   end
 
   def move!
```
Comment (lunks) [inline app/services/order_movement.rb]: Prefer `if` over `unless` and extract these into smaller methods.

File: app/services/produce_unit.rb [hunk 8]
```diff
@@ -6,26 +6,23 @@ def initialize(facility, unit)
...
   end
 
   def produce!
-    Fleet.create(quantity: 1, unit: @unit, squad: @squad, planet: @planet, round: Round.get_current, ready_in: self.ready_in, carrier: @facility) if self.facility? && !self.in_production? && self.metals?
+    Fleet.create(quantity: 1, unit: @unit, squad: @squad, planet: @planet, round: Round.get_current, ready_in: ready_in, carrier: @facility) if valid?
   end
-
-
-
 end
```
Comment (lunks) [inline app/services/produce_unit.rb]: 👏

File: db/schema.rb [hunk 9]
```diff
@@ -10,109 +10,108 @@
 #
```
Comment (lunks) [inline db/schema.rb]: You should skip `schema.rb` on Rubocop because it's autogenerated.

File: app/models/fleet.rb (hunk 1 above)
Comment (rmorali) [reply app/models/fleet.rb]: done

Comment (rmorali) [reply app/models/fleet.rb]: ah, I delegated as :facility?

File: app/models/fleet.rb (hunk 3 above)
Comment (rmorali) [reply app/models/fleet.rb]: Done. much better.

File: app/models/fleet.rb (hunk 4 above)
Comment (rmorali) [reply app/models/fleet.rb]: it do needs carrier! It seems to be a rubocop little misunderstanding. fixed.

File: app/models/squad.rb (hunk 5 above)
Comment (rmorali) [reply app/models/squad.rb]: I prefer ifs too. fixed

File: app/services/order_movement.rb (hunk 7 above)
Comment (rmorali) [reply app/services/order_movement.rb]: fixed

File: app/services/produce_unit.rb (hunk 8 above)
Comment (rmorali) [reply app/services/produce_unit.rb]: thanks to rubocop =P

File: db/schema.rb (hunk 9 above)
Comment (rmorali) [reply db/schema.rb]: I already have done that.

File: app/models/fleet.rb (hunk 2 above)
Comment (rmorali) [reply app/models/fleet.rb]: I am struggling to understand this =P

**Pull request merged**